server/controllers/finance/fiscal.js

```diff
@@ -25,173 +80,95 @@ function getFiscalYears(req, res, next) {
+  if(queryData.start_date){
+    queryData.start_date = new Date(queryData.start_date);
+  }
 
-    return journal.request('create_fiscal_year', periodId, data.user_id, function (error, result) {
-      if (error) {
-        return error;
-      }
-      return result;
-    }, -1, details);
+  var fiscalYearId = req.params.id;
 
-  });
-}
+  var sql = 'UPDATE fiscal_year SET ? WHERE id = ?';
 
-// creates a fiscal year record
-function createNewYear(data) {
-  var sql, monthNo, startMonth, startYear,
-      enterpriseId = data.enterprise_id,
-      startDate = data.start,
-      endDate = data.end,
-      previousFiscalYear = data.previous_fiscal_year || null,
-      fiscalYearText = data.fiscal_year_txt;
-
-  // date math to get the month number, start month, and start year
-  monthNo = monthDiff(startDate, endDate) + 1; // FIXME Why is the plus one?
-  startMonth = startDate.getMonth() + 1;
-  startYear = startDate.getFullYear();
-
-  // template the fiscal year query
-  sql =
-    `INSERT INTO fiscal_year (enterprise_id, number_of_months, fiscal_year_txt, start_month, start_year, previous_fiscal_year) VALUES
-      (?, ?, ?, ?, ?, ?);`;
+  delete queryData.id;
 
-  return db.exec(sql, [enterpriseId, monthNo, fiscalYearText, startMonth, startYear, previousFiscalYear]);
+  lookupFiscalYear(fiscalYearId)
+  .then(function () {
+    return db.exec(sql, [queryData, fiscalYearId]);
+  })
+  .then(function () {
+    return lookupFiscalYear(fiscalYearId);
+  })
+  .then(function (fiscalYear) {
+    res.status(200).json(fiscalYear);
+  })
+  .catch(next)
+  .done();
 }
 
-// creates the periods (including period 0) for a fiscal year
-// TODO -- migrate this to use db.exec() parameter parsing
-function createPeriods(fiscalYearId, start, end) {
-  var sql,
-      totalMonths,
-      periodStart,
-      periodStop,
-      template = [];
-
-  // calculate the total months in the fiscal year to be inserted
-  totalMonths = monthDiff(start, end) + 1;
+/**
+ * Remove a fiscal year details (particularly id)
+ */
 
-  // Initial SQL query without template
-  sql =
-    'INSERT INTO period (fiscal_year_id, period_number, period_start, period_stop) VALUES ';
+function remove(req, res, next) {
 
-  // create the zero period
-  // template.push([fiscalYearId, 0, periodStart, periodStart]);//after discussion, we will not use a period zero
+  var fiscalYearId = req.params.id;
+  var sql = 'DELETE FROM fiscal_year WHERE id = ?';
 
-  // create a period for each month, calculating the
-  // first day and last day of the month
-  for (var i = 0; i < totalMonths; i++) {
-    periodStart = new Date(start.getFullYear(), start.getMonth() + i);
-    periodStop = new Date(periodStart.getFullYear(), periodStart.getMonth() + 1, 0);
-    template.push([fiscalYearId, i+1, periodStart, periodStop]);
-  }
+  lookupFiscalYear(fiscalYearId)
+    .then(function () {
+      return db.exec(sql, [fiscalYearId]);
+    })
+    .then(function () {
+      res.status(204).send();
```

Comment:
A better way to send the status code immediately is `res.sendStatus(204);`.  This ensures that nothing else can be added to the request before it is sent.

See [res.sendStatus()](http://expressjs.com/en/4x/api.html#res.sendStatus) for more information.